www/posts/2020-05-25-v2-release-day.md

```diff
@@ -0,0 +1,152 @@
+
+- O(1) build tooling is easier to reason about, since your build input is always a single file.
+- O(1) build tooling stays fast in large projects because only changed files are ever rebuilt.
+- Caching is much more effective as well, since file changes never invalidate other cached files.
+
+Pair this with a platform like Snowpack that only builds individual files as requested by the browser, and you get a developer experience that's dramatically lighter-weight than your traditional, bundled dev server.
+
+
+## `dev` - A Better Dev Environment
+
+![dev command output example](/img/snowpack-dev-startup-2.png)
+
+Run `snowpack dev` to start your dev environment, and the first thing you'll notice is **how flippin' fast it is.** Snowpack starts up instantly, usually in less than 20ms. 
+
+That's right, 20ms. 
+
+This incredible speed is possible because Snowpack only builds files during development as they are requested by the browser. On your very first page load, Snowpack will build each requested file and then cache it for future use. Because each file is built individually, that cached file can be reused forever without ever needing a second build (until you change the file yourself, that is).
+
+`snowpack dev` includes a dev server for hosting, a fully customizable build pipeline for building, and a bunch more familiar features right out of the box:
+
+- TypeScript Support
+- JSX Support
+- Hot Module Replacement (HMR)
+- Importing CSS
+- Importing CSS Modules
+- Importing Images & Other Assets
+- Custom Routing
+- Proxying Requests
+
+**Build Scripts** are 1-line integrations to connect your favorite build tools into your Snowpack build. Thanks to this new idea of O(1) build tooling, every file build can be expressed as a deterministic `input -> build -> output` workflow, which makes Snowpack perfect for Linux-style piping through an already-existing CLI. 
+
+
+```js
+// snowpack.config.json
+{
+  "scripts": {
+    // Pipe every "*.css" file through the PostCSS CLI
+    // stdin (source file) > postcss > stdout (build output)
+    "build:css": "postcss",
+  }
+}
+```
+
+If you've ever used your `package.json` "scripts" config, Snowpack's build scripts should feel familiar by design. We love how `package.json` scripts are able to use your tooling CLIs directly without forcing you to adopt an unnecessary plugin system, and we tried to recreate that magic feeling here.
+
+**Build Plugins** are even more powerful JavaScript integrations for when a CLI just won't cut it. Maybe your favorite build tool doesn't ship a CLI, or that tool's CLI is too slow, or you just want to write your own integration. Build plugins gives you a clean JavaScript interface to extend your build with.
```

Comment:
CLI is in here 3 times in a row. I would expand the first one to "command line interface", leave the second, and remove the third. Also the double "or" is a bit much.

```
**Build Plugins** are a more powerful JavaScript integration pattern for when a command-line interface won't cut it. Maybe your favorite build tool ships a slow CLI or no CLI. Maybe you just want to write your own integration. Build plugins provide a clean JavaScript interface to extend your build.
```